src/libstd/sys/windows/mod.rs

```diff
@@ -72,10 +72,17 @@ pub fn decode_error_kind(errno: i32) -> ErrorKind {
     }
 }
 
-fn to_utf16_os(s: &OsStr) -> Vec<u16> {
-    let mut v: Vec<_> = s.encode_wide().collect();
-    v.push(0);
-    v
+pub fn to_u16s<S: AsRef<OsStr>>(s: &S) -> io::Result<Vec<u16>> {
+    fn inner(s: &OsStr) -> io::Result<Vec<u16>> {
```

Comment:
I thought this pattern was used to reduce code size, so the outer, essentially no-op function can be inlined, and there are not many (in this case 2) of the monomorphized functions lying around.